librustzcash/src/rustzcash.rs

```diff
@@ -143,9 +143,13 @@ pub extern "system" fn librustzcash_init_zksnark_params(
     let output_path = Path::new(OsStr::from_bytes(unsafe {
         slice::from_raw_parts(output_path, output_path_len)
     }));
-    let sprout_path = Path::new(OsStr::from_bytes(unsafe {
-        slice::from_raw_parts(sprout_path, sprout_path_len)
-    }));
+    let sprout_path = if sprout_path.is_null() && sprout_path_len == 0 {
```

Comment:
```suggestion
    let sprout_path = if sprout_path.is_null() {
```

Per [the `slice::from_raw_parts` documentation](https://doc.rust-lang.org/std/slice/fn.from_raw_parts.html#safety), we must not construct a slice from a null pointer. The current PR would do so if `sprout_path_len != 0`, which is problematic given that this PR is explicitly removing the requirement that the caller ensure this particular pointer is non-null.

Reply:
I had wondered why this was not `if sprout_path.is_null() || sprout_path_len == 0`. If the path length is 0, why try to load it? It may not be necessary, but not knowing the fine details of know null pointers work across ffis, I think this would be the more defensive approach.

Reply:
If the length is zero but the pointer is non-null, this would be interpreted as a zero-length slice, and treated as an error (since it is not a valid path). I think that is more indicative of a mistake by the caller, whereas a null pointer is a much clearer "please do nothing here" indication.